source/routing/raptor_api.cpp

```diff
@@ -76,15 +78,14 @@ static void fill_shape(pbnavitia::Section* pb_section,
                 prev_coord = cur_coord;
             }
         }
-        // Add the coordinates of the stop point if not already added
-        // by the shape.
-        const auto& sp_coord = st->stop_point->coord;
-        if (sp_coord != prev_coord) {
-            add_coord(sp_coord, pb_section);
-            prev_coord = sp_coord;
-        }
         prev_order = cur_order;
     }
+
+    // Adding the coordinates of the last stop point
+    type::GeographicalCoord last_coord = stop_times.back()->stop_point->coord;
```

Comment:
you can keep the const &, no need to copy

Reply:
I changed it.